pkgs/applications/misc/collision/default.nix

```
@@ -10,18 +10,39 @@
 , openssl
 , libxml2
 , pkg-config
+, gitUpdater
+, _experimental-update-script-combinators
+, runCommand
+, crystal2nix
+, writeShellScript
 }:
 crystal.buildCrystalPackage rec {
   pname = "Collision";
-  version = "3.6.0";
+  version = "3.8.0";
 
   src = fetchFromGitHub {
     owner = "GeopJr";
     repo = "Collision";
     rev = "v${version}";
-    hash = "sha256-Qxe4ILDdfYfxu95EvKRTvkAOgDIZDiLymBlZouBWn0M=";
+    hash = "sha256-Bo/u0UYM/N7tLqdCs2OU5pdj2s9LXPooSR1PCGk9dSc=";
   };
-  patches = [ ./make.patch ];
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace-fail './bin/gi-crystal || $(CRYSTAL_LOCATION)shards install && ./bin/gi-crystal' gi-crystal \
+      --replace-fail 'gtk-update-icon-cache $(PREFIX)/share/icons/hicolor' 'true'
+  '';
+
+  XMLLINT = "${libxml2}/bin/xmllint";
+
+  postConfigure = ''
+    rm lib/gi-crystal
+    cp -rL lib lib-mut
+    chmod -R +w lib-mut
+    rm -rf lib
+    mv lib-mut lib
+  '';
+
   shardsFile = ./shards.nix;
 
```

I discussed about gi-crystal in this [PR](https://github.com/NixOS/nixpkgs/pull/305484)
Shortly, using gi-crystal as a nix dependency is a bad idea, because gi-crystal is not backwards compatible and its version must match the version in the shard.lock
But buildCrystalPackage doesn't run `shards install` command, so you need to build it manually
```suggestion
  preBuild = ''
    cd lib/gi-crystal && shards build -Dpreview_mt --release --no-debug
    cd ../.. && mkdir bin/ && cp lib/gi-crystal/bin/gi-crystal bin/
  '';
```

---

Done